features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/ecp/ecp_internal_alt.c

```diff
@@ -0,0 +1,807 @@
+     */
+
+    if (modop == MODOP_MUL ||
+        modop == MODOP_ADD ||
+        modop == MODOP_SUB) {
+        INTERNAL_MPI_NORM(&Np, *o1, N_, *p);
+        MBEDTLS_MPI_CHK(internal_mpi_write_eccreg(Np, (uint32_t *) CRPT->ECC_X1, NU_ECC_BIGNUM_MAXWORD));
+        INTERNAL_MPI_NORM(&Np, *o2, N_, *p);
+        MBEDTLS_MPI_CHK(internal_mpi_write_eccreg(Np, (uint32_t *) CRPT->ECC_Y1, NU_ECC_BIGNUM_MAXWORD));
+    } else if (modop == MODOP_DIV) {
+        INTERNAL_MPI_NORM(&Np, *o2, N_, *p);
+        MBEDTLS_MPI_CHK(internal_mpi_write_eccreg(Np, (uint32_t *) CRPT->ECC_X1, NU_ECC_BIGNUM_MAXWORD));
+        INTERNAL_MPI_NORM(&Np, *o1, N_, *p);
+        MBEDTLS_MPI_CHK(internal_mpi_write_eccreg(Np, (uint32_t *) CRPT->ECC_Y1, NU_ECC_BIGNUM_MAXWORD));
+    } else {
+        MBEDTLS_MPI_CHK(MBEDTLS_ERR_ECP_BAD_INPUT_DATA);
+    }
+
+    MBEDTLS_MPI_CHK(internal_mpi_write_eccreg(p, (uint32_t *) CRPT->ECC_N, NU_ECC_BIGNUM_MAXWORD));
+
+    crypto_ecc_prestart();
+    CRPT->ECC_CTL = (pbits << CRPT_ECC_CTL_CURVEM_Pos) | (ECCOP_MODULE | modop) | CRPT_ECC_CTL_FSEL_Msk | CRPT_ECC_CTL_START_Msk;
+    ecc_done = crypto_ecc_wait();
+    
+    /* FIXME: Better error code for ECC accelerator error */
+    MBEDTLS_MPI_CHK(ecc_done ? 0 : -1);
+    
+    /* X1 holds the result. */
+    MBEDTLS_MPI_CHK(internal_mpi_read_eccreg(r, (uint32_t *) CRPT->ECC_X1, NU_ECC_BIGNUM_MAXWORD));
+
+cleanup:
+    
+    mbedtls_mpi_free(&N_);
+
+    return ret;
+}
+
+#endif  // ECP_SHORTWEIERSTRASS
+
+NU_STATIC int internal_mpi_read_eccreg(mbedtls_mpi *x, const volatile uint32_t *eccreg, size_t eccreg_num)
+{
+    if (x == NULL) {
+        return MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+    }
+    
+    int ret;
+    size_t i, n;
+
+    for (n = eccreg_num; n > 0; n --) {
+        if (eccreg[n - 1] != 0) {
+            break;
+        }
+    }
+    
+    MBEDTLS_MPI_CHK(mbedtls_mpi_lset(x, 0));
+    MBEDTLS_MPI_CHK(mbedtls_mpi_grow(x, WORDS_TO_LIMBS(n)));
+
+    for (i = 0; i < n; i ++) {
+        x->p[i / wiL] |= ((mbedtls_mpi_uint) eccreg[i]) << ((i % wiL) << 5);
+    }
+
+cleanup:
+
+    return ret;
+}
+
+NU_STATIC int internal_mpi_write_eccreg( const mbedtls_mpi *x, volatile uint32_t *eccreg, size_t eccreg_num )
+{
+    if (x == NULL) {
+        return MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+    }
+    
+    if (mbedtls_mpi_cmp_int(x, 0) < 0) {
+        return MBEDTLS_ERR_MPI_NEGATIVE_VALUE;
+    }
+    
+    size_t i, n;
+
+    /* How many words needed? */
+    n = (mbedtls_mpi_size(x) + sizeof (uint32_t) - 1) / sizeof (uint32_t);
+
+    if (eccreg_num < n) {
+        return MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL;
+    }
+
+    /* Fill non-zero part */
+    for (i = 0; i < n; i ++) {
+        eccreg[i] = (uint32_t) (x->p[i / wiL] >> ((i % wiL) << 5));
+    }
+    
+    /* Zeroize remaining part
+     *
+     * crypto_zeroize32() has excluded optimization doubt, so we can safely set H/W registers to 0 via it.
+     */
+    crypto_zeroize32((uint32_t *) eccreg + i, eccreg_num - i);
```

Comment:
Could you change this to `n` instead of `i`, please?

Reply:
@hanno-arm I fixed it in https://github.com/ARMmbed/mbed-os/pull/5812/commits/03f0ea11175dff7b435e43ac0242ce7da7c90caf.